### PR TITLE
Issue #2029: Address Hint Bar Blockers

### PR DESCRIPTION
**Pull request by Baron-Severin**
Per UX: https://github.com/mozilla-mobile/firefox-tv/issues/1907#issuecomment-481074681

### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
- [x] This PR includes a **CHANGELOG entry** or does not need one
- [x] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

**Comment by Baron-Severin** [comment]
Note that I wasn't able to set up the experiment yet.  I've opened a ticket requesting access to the Fretboard back end.

File: app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt [hunk 1]
```diff
@@ -204,14 +204,14 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
             }.forceExhaustive
         }.addTo(compositeDisposable)
 
-        val (hintViewModel, progressBarGravity) = if (serviceLocator!!.experimentsProvider.shouldShowHintBar()) {
+        val (hintViewModel, progressBarBottomMargin) = if (serviceLocator!!.experimentsProvider.shouldShowHintBar()) {
             FirefoxViewModelProviders.of(this).get(WebRenderHintViewModel::class.java) to
-                    (Gravity.END or Gravity.BOTTOM)
+                    resources.pxToDp(64)
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt]: +1 on the link to the UX comment

File: app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderHintViewModel.kt [hunk 2]
```diff
@@ -31,57 +31,47 @@ class WebRenderHintViewModel(
...
                 .autoConnect(0)
     }
     override val hints: Observable<List<HintContent>> = Observable.just(listOf(openMenuHint))
 
-    /**
-     * Emits true when the hint bar should be shown, or false when it should be hidden
-     */
-    private val webRenderOpenedEvents by lazy {
-        screenController.currentActiveScreen
+    private val webRenderOpened = screenController.currentActiveScreen
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderHintViewModel.kt]: Why not lazy anymore?

File: app/src/main/java/org/mozilla/tv/firefox/experiments/ExperimentsProvider.kt [hunk 3]
```diff
@@ -46,24 +46,17 @@ class ExperimentsProvider(private val fretboard: Fretboard, private val context:
...
-//            expDescriptor.name.endsWith(ExperimentSuffix.B.value) -> false
-//            else -> {
-//                Sentry.capture(ExperimentIllegalStateException("Hint Bar Illegal Branch Name"))
-//                false
-//            }
-//        }
+        val expDescriptor = checkBranchVariants(ExperimentConfig.HINT_BAR_TEST)
+        return when {
+            // The user is currently not part of the experiment
+            expDescriptor == null -> false
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/experiments/ExperimentsProvider.kt]: Should we make this "not in experiment" handling consistent between this and AA test? Though I guess we'll be removing it anyway.

File: app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderHintViewModel.kt (hunk 2 above)
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderHintViewModel.kt]: I'm using lazy when the compiler gets unhappy that I'm declaring variables out of order (I want to organize top to bottom on importance, it wants them all declared before usage).  It's not a great solution, but it's the best I've come up with so far.

With the change, this one didn't need that anymore.

File: app/src/main/java/org/mozilla/tv/firefox/experiments/ExperimentsProvider.kt (hunk 3 above)
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/tv/firefox/experiments/ExperimentsProvider.kt]: Personally, I find nested conditionals confusing.  If I were to change either here, I would prefer to flatten the A/A.

**Comment by Baron-Severin** [comment]
@liuche 
> What else needs to be done on the Experimental side?

1) set up dashboard, 2) update experiment (currently it's at 0% show and 100% hide)

**Pull request merged**